components/home/calendar-events.tsx

```diff
@@ -1,38 +1,42 @@
-interface CalendarData {
-    dayNumber: number;
-    suffix: string;
-    day: string;
-    title: string;
-    month: string;
-}
+import dayjs from "dayjs";
+import { calendar_v3 } from "googleapis";
+import advancedFormat from "dayjs/plugin/advancedFormat";
+import { useEffect } from "react";
 
-interface CalendarEventsProps {
-    data: CalendarData[];
-}
+export function CalendarEvents({ data }: { data: calendar_v3.Schema$Event[] }) {
+  useEffect(() => {
+    dayjs.extend(advancedFormat);
```

Comment:
I haven't seen anyone do that before. Why do we need to extend dayjs inside the useEffect rather than outside the component as a whole?

Reply:
oh, i just thought this was better since the useEffect will only run on load instead of running multiple times but I guess doing it outside of the component will achieve the same thing and is more common so I'll change it